akka-docs-dev/rst/java/stream-cookbook.rst

```diff
@@ -107,23 +107,30 @@ Implementing reduce-by-key
 elements.
 
 The "hello world" of reduce-by-key style operations is *wordcount* which we demonstrate below. Given a stream of words
-we first create a new stream ``wordStreams`` that groups the words according to the ``i -> i`` function, i.e. now
+we first create a new stream that groups the words according to the ``i -> i`` function, i.e. now
 we have a stream of streams, where every substream will serve identical words.
 
-To count the words, we need to process the stream of streams (the actual groups containing identical words). By mapping
-over the groups and using ``fold`` (remember that ``fold`` automatically materializes and runs the stream it is used
-on) we get a stream with elements of ``Future[String,Int]``. Now all we need is to flatten this stream, which
-can be achieved by calling ``mapAsync`` with ``i -> i`` identity function.
-
-There is one tricky issue to be noted here. The careful reader probably noticed that we put a ``buffer`` between the
-``mapAsync()`` operation that flattens the stream of futures and the actual stream of futures. The reason for this is
-that the substreams produced by ``groupBy()`` can only complete when the original upstream source completes. This means
-that ``mapAsync()`` cannot pull for more substreams because it still waits on folding futures to finish, but these
-futures never finish if the additional group streams are not consumed. This typical deadlock situation is resolved by
-this buffer which either able to contain all the group streams (which ensures that they are already running and folding)
-or fails with an explicit failure instead of a silent deadlock.
-
-.. includecode:: ../../../akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/cookbook/RecipeReduceByKey.java#word-count
+To count the words, we need to process the stream of streams (the actual groups
+containing identical words). ``GroupBy`` returns a :class:`SubSource`, which
```

Comment:
weird capitalization of GroupBy since in code it is small-caps. I usually try to refactor these kind of sentences so that the first word is not code to sidestep this issue.